src/UserSpaceInstrumentation/AllocateInTraceeTest.cpp

```diff
@@ -170,4 +170,64 @@ TEST(AllocateInTraceeTest, AutomaticAllocateAndFree) {
   waitpid(pid, nullptr, 0);
 }
 
+TEST(AllocateInTraceeTest, ReadSeccompModeOfThread) {
+  std::optional<int> seccomp_mode = ReadSeccompModeOfThread(getpid());
+  ASSERT_TRUE(seccomp_mode.has_value());
+  EXPECT_THAT(seccomp_mode.value(),
+              testing::AnyOf(SECCOMP_MODE_DISABLED, SECCOMP_MODE_STRICT, SECCOMP_MODE_FILTER));
+}
+
+TEST(AllocateInTraceeTest, SyscallInTraceeFailsBecauseOfSeccomp) {
+  pid_t pid = fork();
+  ORBIT_CHECK(pid != -1);
+  if (pid == 0) {
+    prctl(PR_SET_PDEATHSIG, SIGTERM);
+
+    // Transition to strict seccomp mode.
+    int64_t seccomp_result = syscall(SYS_seccomp, SECCOMP_SET_MODE_STRICT, 0, nullptr);
+    ORBIT_CHECK(seccomp_result == 0);
+
+    // Child just runs an endless loop.
+    volatile uint64_t counter = 0;
+    while (true) {
+      // Endless loops without side effects are UB and recent versions of clang optimize it away.
+      ++counter;
+    }
+  }
+
+  // Give the child enough time to execute the seccomp syscall.
+  std::this_thread::sleep_for(std::chrono::milliseconds{10});
```

Comment:
That would be an effective explanation. I had already increased from 1 ms (which works locally) so I didn't think this could be an issue. I'll start with trying what happens by increasing even more and then we go from there.